src/service/server.rs

```diff
@@ -19,60 +20,68 @@ use crate::util::error::{quit_error, ErrorHints};
 ///
 /// Spawns a tokio runtime to complete all work on.
 #[tokio::main(flavor = "multi_thread")]
-pub async fn service(config: Arc<Config>) -> Result<(), ()> {
-    // Load server state
-    let server = Arc::new(Server::default());
-
+pub async fn service(bind_addr: SocketAddr, configs: Vec<Arc<Config>>) -> Result<(), ()> {
     // Listen for new connections
-    let listener = TcpListener::bind(config.public.address)
-        .await
-        .map_err(|err| {
-            quit_error(
-                anyhow!(err).context("Failed to start proxy server"),
-                ErrorHints::default(),
-            );
-        })?;
+    let listener = TcpListener::bind(bind_addr).await.map_err(|err| {
+        quit_error(
+            anyhow!(err).context("Failed to start proxy server"),
+            ErrorHints::default(),
+        );
+    })?;
 
-    info!(
-        target: "lazymc",
-        "Proxying public {} to server {}",
-        config.public.address, config.server.address,
-    );
+    let mut servers = HashMap::new();
 
-    if config.lockout.enabled {
-        warn!(
+    for config in configs.iter() {
```

Comment:
We should probably extract this (the logic for doing something for many configuration files) into a separate function we can invoke for each configuration.

This is now a bit messy (and I admit my code already was).